config/initializers/rss_notifications.rb

```diff
@@ -0,0 +1,10 @@
+# First, we check if there's a job already enqueued for RSS notifications
+pending_jobs = ::Foreman::Application.dynflow.world.persistence.find_execution_plans(filters: { :state => 'scheduled' })
+scheduled_job = pending_jobs.select do |job|
+  delayed_plan = ::Foreman::Application.dynflow.world.persistence.load_delayed_plan job.id
+  next unless delayed_plan.present?
+  delayed_plan.to_hash[:serialized_args].first["job_class"] == 'CreateRssNotifications'
+end
+
+# Only create notifications if there isn't a scheduled job
+CreateRssNotifications.perform_later unless scheduled_job.present?
```

Comment:
is this the best way to load future / scheduled tasks? I was hoping this can be kept in somewhere and load all of them as one process.

Also, shouldnt this done on dynflow load vs Foreman load? /cc @ares and @iNecas

Reply:
Not sure about 'best way', but it's a way. As far as I know this instruction is still the same process as Foreman (it's just enqueing the job in Dynflow),

I don't think anything should be done on Dynflow load, I prefer to treat it as a background processing engine completely detached from Foreman, like Resque, Sidekiq, etc.. -

Reply:
it would be imho a good approach to have a separate service object - scheduler, that would plan other jobs and keep an eye on them, it would be regularly called using dynflow job but I think we can add it later when we have more repetitive jobs. This is good for now and and easy to replace later.